### PR TITLE
v2 to v3 fragment changes for parseClusterFromV2Yaml

### LINKED ISSUE #10843: Convert v2 API test fragments to v3
While all programatically constructed test configs will be v3 (via API boosting rewriting), there are still v2 JSON/YAML fragments in test code. Many of these will "just work" as v3, but some may have v2 features that will have them treated as v2. This becomes confusing when they are extended by developers who want to model v3 features.

These need to be audited and updated as appropriate. Also, the names of methods such as `parseAccessLogFromV2Yaml` should be updated to reflect v3 when operating on fragments that should be v3.

This is somewhat similar to the work done by @dmitri-d for examples in https://github.com/envoyproxy/envoy/pull/10644 as part of https://github.com/envoyproxy/envoy/issues/9735.

CC @alyssawilk

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: v2 to v3 fragment changes for parseClusterFromV2Yaml
Additional Description:
Risk Level: low
Testing: unit, format and integration test
Docs Changes: NA
Release Notes:
[Optional Runtime guard:]
Optional Fixes #10843 
[Optional Deprecated:]

### CODE REVIEW

**Comment by ankatare** [comment]
need help to understand errors in current build issues. seems some issue in java side to download packages. 

logs :
java.io.IOException: Error downloading [https://mirror.bazel.build/openjdk/azul-zulu11.37.17-ca-jdk11.0.6/zulu11.37.17-ca-jdk11.0.6-linux_x64.tar.gz] to /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/external/remotejdk11_linux/zulu11.37.17-ca-jdk11.0.6-linux_x64.tar.gz: Checksum was 2c749bea0c7362b7e2bfc34ceb82111ed2298ab2be9ccfba15ba71adc5293f30 but wanted 360626cc19063bc411bfed2914301b908a8f77a7919aaea007a977fa8fb3cde1
ERROR: /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/external/bazel_tools/tools/jdk/BUILD:492:1: @bazel_tools//tools/jdk:remote_jdk11 depends on @remotejdk11_linux//:jdk in repository @remotejdk11_linux which failed to fetch. no such package '@remotejdk11_linux//': java.io.IOException: Error downloading [https://mirror.bazel.build/openjdk/azul-zulu11.37.17-ca-jdk11.0.6/zulu11.37.17-ca-jdk11.0.6-linux_x64.tar.gz] to /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/external/remotejdk11_linux/zulu11.37.17-ca-jdk11.0.6-linux_x64.tar.gz: Checksum was 2c749bea0c7362b7e2bfc34ceb82111ed2298ab2be9ccfba15ba71adc5293f30 but wanted 360626cc19063bc411bfed2914301b908a8f77a7919aaea007a977fa8fb3cde1

File: test/common/upstream/logical_dns_cluster_test.cc [hunk 1]
```diff
@@ -386,7 +399,7 @@ TEST_F(LogicalDnsClusterTest, BadConfig) {
       port_value: 443
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(setupFromV2Yaml(multiple_hosts_yaml), EnvoyException,
+  EXPECT_THROW_WITH_MESSAGE(setupFromV3Yaml(multiple_hosts_yaml, false), EnvoyException,
```
Comment (htuch) [inline test/common/upstream/logical_dns_cluster_test.cc:402]: I don't think all tests with exceptions should avoid boosting, only those where the exception is relate to some deprecated feature.

File: test/common/upstream/original_dst_cluster_test.cc [hunk 2]
```diff
@@ -136,7 +138,7 @@ TEST_F(OriginalDstClusterTest, BadConfigWithLoadAssignment) {
   )EOF";
 
   EXPECT_THROW_WITH_MESSAGE(
-      setupFromYaml(yaml), EnvoyException,
+      setupFromYaml(yaml, false), EnvoyException,
```
Comment (htuch) [inline test/common/upstream/original_dst_cluster_test.cc:141]: Ditto.

File: test/common/upstream/upstream_impl_test.cc [hunk 3]
```diff
@@ -2231,7 +2397,7 @@ TEST_F(ClusterInfoImplTest, OneofExtensionProtocolOptionsForUnknownFilter) {
         "@type": type.googleapis.com/google.protobuf.Struct
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(makeCluster(yaml), EnvoyException,
+  EXPECT_THROW_WITH_MESSAGE(makeCluster(yaml, false), EnvoyException,
```
Comment (htuch) [inline test/common/upstream/upstream_impl_test.cc:2400]: Ditto, etc.

File: test/common/upstream/original_dst_cluster_test.cc (hunk 2 above)
Comment (ankatare) [reply test/common/upstream/original_dst_cluster_test.cc:141]: @htuch done, thanks for review it.

File: test/common/upstream/logical_dns_cluster_test.cc (hunk 1 above)
Comment (ankatare) [reply test/common/upstream/logical_dns_cluster_test.cc:402]: @htuch thanks for review. done.
 here i need to update exception message according to V3 exception output.

File: test/common/upstream/upstream_impl_test.cc (hunk 3 above)
Comment (ankatare) [reply test/common/upstream/upstream_impl_test.cc:2400]: @htuch in my view this also required boosting as field "extension_protocol_options " is not defined in V3 and this particular test case required both to present (typed_extension_protocol_options  & extension_protocol_options )and then test.
see exception output "Only one of typed_extension_protocol_options or extension_protocol_options can be specified"

Comment (ankatare) [reply test/common/upstream/upstream_impl_test.cc:2400]: @htuch similar case for few more test cases where those test cases particularly required "extension_protocol_options " field to be present to test functionalities, so boosting is preferred. please suggest.

**Comment by ankatare** [comment]
seems some unexpected issue in CI build. need help to fix this.

File: test/common/upstream/upstream_impl_test.cc (hunk 3 above)
Comment (htuch) [reply test/common/upstream/upstream_impl_test.cc:2400]: I'd suggest updating the test to use `typed_extension_protocol_options` instead.

File: test/common/upstream/original_dst_cluster_test.cc [hunk 4]
```diff
@@ -517,7 +519,7 @@ TEST_F(OriginalDstClusterTest, UseHttpHeaderEnabled) {
 
   EXPECT_CALL(initialized_, ready());
   EXPECT_CALL(*cleanup_timer_, enableTimer(_, _));
-  setupFromYaml(yaml);
+  setupFromYaml(yaml, false);
```
Comment (htuch) [inline test/common/upstream/original_dst_cluster_test.cc:522]: Why boost here?

File: test/common/upstream/upstream_impl_test.cc (hunk 3 above)
Comment (htuch) [reply test/common/upstream/upstream_impl_test.cc:2400]: Actually, I see what's going on. This one is OK.

File: test/common/upstream/upstream_impl_test.cc [hunk 5]
```diff
@@ -2564,26 +2746,26 @@ TEST_F(ClusterInfoImplTest, ExtensionProtocolOptionsForFilterWithOptions) {
     TestNetworkFilterConfigFactory factory(factoryBase);
     Registry::InjectFactory<Server::Configuration::NamedNetworkFilterConfigFactory> registry(
         factory);
-    clusters.push_back(makeCluster(yaml));
+    clusters.push_back(makeCluster(yaml, false));
```
Comment (htuch) [inline test/common/upstream/upstream_impl_test.cc:2749]: I think these tests _should_ be converted to typed extension protocol options, there shouldn't be a need to boost though.

File: test/common/upstream/original_dst_cluster_test.cc (hunk 4 above)
Comment (ankatare) [reply test/common/upstream/original_dst_cluster_test.cc:522]: @htuch : my fault. i corrected it.

File: test/common/upstream/upstream_impl_test.cc (hunk 5 above)
Comment (ankatare) [reply test/common/upstream/upstream_impl_test.cc:2749]: @htuch made changes as per your suggestion, let me know if this is not ok.   anything else apart from this particular test case you want to convert in typed extension protocol options ?

File: test/common/upstream/upstream_impl_test.cc [hunk 6]
```diff
@@ -2457,6 +2630,7 @@ struct TestFilterProtocolOptionsConfig : public Upstream::ProtocolOptionsConfig
 
 // Cluster extension protocol options fails validation when configured for filter that does not
 // support options.
+// This test case can't be converted for V3 API as it is specific for extension_protocol_options
```
Comment (htuch) [inline test/common/upstream/upstream_impl_test.cc:2633]: Can you convert all the tests to the new typed variant for extension protocol options? Ideally we only have the essential cases still doing boosting.

Comment (ankatare) [reply test/common/upstream/upstream_impl_test.cc:2633]: @htuch done. thanks for your review.

**Pull request merged**